x/icqoracle/types/genesis.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	fmt "fmt"
+)
+
+// DefaultGenesis returns the default genesis state
+func DefaultGenesis() *GenesisState {
+	return &GenesisState{}
+}
+
+// Performs basic genesis state validation by iterating through all token prices and validating
+// using ValidateBasic() since it already implements thorough validation of the important token
+// price fields.
+// We ignore the SpotPrice, UpdatedAt & QueryInProgress fields since they are reset in InitGenesis().
+func (gs GenesisState) Validate() error {
+	for i, tokenPrice := range gs.TokenPrices {
+
+		msg := NewMsgRegisterTokenPriceQuery(
+			"stride1palmssweatykneesweakarmsareheavy8ahm9u", // dummy address, not stored in token price
+			tokenPrice.BaseDenom,
+			tokenPrice.QuoteDenom,
+			tokenPrice.OsmosisPoolId,
+			tokenPrice.OsmosisBaseDenom,
+			tokenPrice.OsmosisQuoteDenom,
+		)
```

Comment:
@sampocs sampocs Dec 17, 2024

this is kinda hacky haha, you could just separate out the checks into its own function that's called by both this and validate basic

Reply:
@sampocs 4188ddafa